src/draw.c

```diff
@@ -176,6 +176,15 @@ static int get_text_icon_padding()
         }
 }
 
+static int get_vertical_text_icon_padding()
+{
+        if (settings.text_icon_padding) {
+                return settings.text_icon_padding;
```

Comment:
I believe the text_icon_padding should only be used when the icon is at the top. If the icon icon is not top, you should just use normal vertical padding.
Same issue in `get_horizontal_text_icon_padding()`.

This makes me notice that the code already had the issue that when the icon is off, the `text_icon_padding` is still used. Could you change that as well?

Reply:
Do you have an opinion on adding a `vertical_padding` setting? I would imagine the `padding` setting would then just serve as a default if either vertical or horizontal are unset

Reply:
> I believe the text_icon_padding should only be used when the icon is at the top. If the icon icon is not top, you should just use normal vertical padding.
Same issue in get_horizontal_text_icon_padding().

I attempted to address this in the latest commits

> This makes me notice that the code already had the issue that when the icon is off, the `text_icon_padding` is still used. Could you change that as well?

I wasn't able to reproduce this. Are you able to point out which lines this is happening on?

Reply:
> Do you have an opinion on adding a vertical_padding setting? I would imagine the padding setting would then just serve as a default if either vertical or horizontal are unset

The padding variable _is_ the vertical padding currently. It just has an unfortunate name. I think it would not be too breaking to make it the general padding setting and create a separate vertical padding setting.

Reply:
>   I wasn't able to reproduce this. Are you able to point out which lines this is happening on?

I might have been wrong on this one, I didn't test if it would happen. It's probably handled well already by just ignoring the icon padding when no icon is present.

Reply:
>The padding variable is the vertical padding currently. It just has an unfortunate name. I think it would not be too breaking to make it the general padding setting and create a separate vertical padding setting.

Would replacing accesses of `settings.h_padding` and `settings.padding` with these be a reasonable approach?

```C
static int get_horizontal_padding()
{
        if (settings.h_padding) {
                return settings.h_padding;
        } else {
                return settings.padding;
        }
}

static int get_vertical_padding()
{
        if (settings.v_padding) {
                return settings.v_padding;
        } else {
                return settings.padding;
        }
}
```

Alternatively I could make it so `settings.h_padding` or `settings.v_padding` are set to `settings.padding` if they are unset when the settings object gets created, so no check needs to be performed and everything can just use `settings.{h/v}_padding`

Reply:
I think the first approach is best, since that would open the possibility of making the padding settings rules later. I would recommend you do this addition in a separate PR, though, to keep this one from getting too big.